### PR TITLE
Add settings option for status bar decoration in debugging session

### LINKED ISSUE #155294: Allow me to disable debug status bar treatment
I am becoming a power-user of our debugger (yeah, finally). This means that most of my "VS Code time" I have some kind of debuggee running or connected. So, I am seeing the ugly orange debug status bar treatment a lot. I don't like it and for me it's enough to see the activitybar icon decorations. So, pretty please give me an option to disable this

### PR DESCRIPTION
**Pull request by KoushikSahu**
This PR fixes #155294

cc: @roblourens

### CODE REVIEW

**Comment by KoushikSahu** [comment]
@roblourens Can you please review my PR? Also I believe you would be able to provide necessary permission to run the remaining gh actions.

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts [hunk 1]
```diff
@@ -581,6 +581,11 @@ configurationRegistry.registerConfiguration({
 			type: 'boolean',
 			default: false,
 			description: nls.localize('debug.autoExpandLazyVariables', "Automatically show values for variables that are lazily resolved by the debugger, such as getters.")
+		},
+		'debug.DecorateStatusBar': {
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:585]: Maybe `debug.enableStatusBarColor`?

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts [hunk 2]
```diff
@@ -581,6 +581,11 @@ configurationRegistry.registerConfiguration({
...
+			type: 'boolean',
+			description: nls.localize({ comment: ['This is the description for a setting'], key: 'decorateStatusBar' }, "Decorate status bar when debugger is connected."),
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:587]: Comment is not necessary

Comment (roblourens) [reply src/vs/workbench/contrib/debug/browser/debug.contribution.ts:587]: "Color status bar when debugger is active"

File: src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts [hunk 3]
```diff
@@ -61,15 +62,17 @@ export class StatusBarColorProvider implements IWorkbenchContribution {
...
 	) {
 		this.debugService.onDidChangeState(this.update, this, this.disposables);
 		this.contextService.onDidChangeWorkbenchState(this.update, this, this.disposables);
 		this.update();
 	}
 
 	protected update(): void {
-		this.enabled = isStatusbarInDebugMode(this.debugService.state, this.debugService.getViewModel().focusedSession);
+		const decorateStatusBar: boolean = this.configurationService.getValue<IDebugConfiguration>('debug').decorateStatusBar;
+		if (decorateStatusBar) { this.enabled = isStatusbarInDebugMode(this.debugService.state, this.debugService.getViewModel().focusedSession); }
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts:75]: Should be formatted on 3 lines (running "Format Document" might fix this)

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts (hunk 1 above)
Comment (KoushikSahu) [reply src/vs/workbench/contrib/debug/browser/debug.contribution.ts]: lgtm

**Comment by KoushikSahu** [comment]
@roblourens thanks for the review. Let me know if this works.

**Review by roblourens — CHANGES_REQUESTED** [review]
Actually, this needs a little more. `update()` should run immediately when the setting changes. I should be able to toggle this even while debugging and see the effect. And currently if I disable it while in a debugging session, then the color never goes away. So `this.enabled` should always be set to something when update runs, instead of skipping it. Please test it a little bit like this and make sure nothing weird can happen.

**Comment by KoushikSahu** [comment]
> Actually, this needs a little more. `update()` should run immediately when the setting changes. I should be able to toggle this even while debugging and see the effect. And currently if I disable it while in a debugging session, then the color never goes away. So `this.enabled` should always be set to something when update runs, instead of skipping it. Please test it a little bit like this and make sure nothing weird can happen.

@roblourens Yeah I didn't test it this way. I'll try and figure this out.

**Comment by KoushikSahu** [comment]
> Actually, this needs a little more. `update()` should run immediately when the setting changes. I should be able to toggle this even while debugging and see the effect. And currently if I disable it while in a debugging session, then the color never goes away. So `this.enabled` should always be set to something when update runs, instead of skipping it. Please test it a little bit like this and make sure nothing weird can happen.

@roblourens So I made sure that the `this.enable` is always set to some boolean when `update` function is called. The behaviour I got was the same as before. During the debugging session if user changes the setting then the color is not reflected immediately but after performing some debugging action (like step over, step up ...).

**Comment by roblourens** [comment]
You can listen to `configurationService.onDidChangeConfiguration` and update when that setting is changed.

**Comment by KoushikSahu** [comment]
Been busy this month. I have time available now. I will be working on the required code changes from now.

**Comment by jrieken** [comment]
Did this go stale? Iff so I'll work on a PR

**Comment by KoushikSahu** [comment]
> Did this go stale? Iff so I'll work on a PR

Not getting the time to work on this. Sorry for the inconvenience. Please take this issue up.

**Pull request closed without merging**